aspnet/mvc/models/validation.rst

```diff
@@ -1,8 +1,196 @@
+   :language: c#
+   :lines: 5-14
+   :dedent: 4
+  
+Custom validation
+-----------------
+Data annotations work for most validation needs. However, some validation rules are specific to your business, as they're not just generic data validation such as ensuring a field is required or that it conforms to a range of values. For these scenarios, custom validation attributes are a great solution. Creating your own custom validation attributes in MVC is easy. Just inherit from the ``ValidationAttribute``, and override the ``IsValid`` method. The ``IsValid`` method accepts two parameters, the first is an object named `value` and the second is a ``ValidationContext`` object named `validationContext`. `Value` refers to the actual value from the field that your custom validator is validating. 
+
+In the following sample, a business rule that states that users may not set the genre to `Classic` for a movie released after 1960. The ``[ClassicMovie]`` attribute checks the genre first, and if it is a classic, then it checks the release date to see that it is later than 1960. If it is released after 1960, validation fails. The attribute accepts an integer parameter representing the year that you can use to validate data. You can capture the value of the parameter in the attribute's constructor, as shown here:
+					 
+.. literalinclude:: validation/sample/ClassicMovieAttribute.cs
+   :language: c#
+   :lines: 11-37
+   :dedent: 4
+   
+The ``movie`` variable above represents a ``Movie`` object that contains the data from the form submission to validate. In this case, the validation code checks the date and genre in the ``IsValid`` method of the ``ClassicMovieAttribute`` class as per the rules. Upon successful validation ``IsValid`` returns a ``ValidationResult.Success`` code, and when validation fails, a ``ValidationResult`` with an error message. When a user modifies the ``Genre`` field and submits the form, the ``IsValid`` method of the ``ClassicMovieAttribute`` will verify whether the movie is a classic. Like any built-in attribute, apply the ``ClassicMovieAttribute`` to a property such as ``ReleaseDate`` to ensure validation happens, as shown in the previous code sample. 
+
+Alternatively, this same code could be placed in the model instead by implementing ``IValidatableObject`` which contains a ``Validate`` method. This works well for one-off validations like in the sample.
+
+ .. literalinclude:: validation/sample/MovieIValidatable.cs
+   :language: c#
+   :lines: 35-48
+   :dedent: 8
+  
+Client side validation
+----------------------
+Client side validation is a great convenience for users. It saves time they would otherwise spend waiting for a round trip to the server. In business terms, even a few seconds multiplied hundreds of times each day adds up to be a lot of time, expense, and frustration. Straightforward and immediate validation enables users to work more efficiently and produce better quality input and output. 
```

Comment:
Client side validation is primarily for the user - immediate feedback when they leave each input element.  A form with many inputs doesn't need to wait on the post.